### PR TITLE
Fix C99 construct without -c99

### PR DESCRIPTION
**Pull request by mscherer**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [X] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

See previous PR (#16850 #14593), this doesn't build unless -c99 is used for gcc.

### CODE REVIEW

**Comment by karliss** [comment]
Seems a bit weird that gcc complains about variable declaration in for loop but is fine with variable declaration not being at the top of scope. Both of them are c99 features.

File: libr/bin/dwarf.c [hunk 1]
```diff
@@ -424,7 +424,8 @@ static const ut8 *parse_line_header (
 		if (f) {
 			fprintf (f, " Opcodes:\n");
 		}
-		for (int i = 1; i <= hdr->opcode_base - 1; i++) {
+		int i;
```
Comment (trufae) [inline libr/bin/dwarf.c:427]: use size_t

File: libr/bin/dwarf.c [hunk 2]
```diff
@@ -424,7 +424,8 @@ static const ut8 *parse_line_header (
...
+		for (i = 1; i <= hdr->opcode_base - 1; i++) {
```
Comment (trufae) [inline libr/bin/dwarf.c:428]: use i < hdr_opcode_base instead of <= and -1

File: libr/bin/dwarf.c (hunk 1 above)
Comment (trufae) [reply libr/bin/dwarf.c:427]: also replace <=-1 with <

**Pull request merged**